golem-rib/src/text/mod.rs

```diff
@@ -425,6 +425,25 @@ mod record_tests {
         let output_expr = from_string(expr_str.as_str()).unwrap();
         assert_eq!((expr_str, input_expr), (expected_record_str, output_expr));
     }
+
+    #[test]
+    fn test_round_trip_record_literal_invalid() {
+        let expr_str = r#"
+                 ${  {body: golem:component/api.{get-character}(), headers: { x-test: 'foobar' } } }
```

Comment:
The existence of this test is not a big problem, however I wouldn't personally include this test if was doing it. So its very opinionated statement.
 
All other tests in golem-rib is structured (as much as I can) and has a behavior. Example: testing a  `record`parser will test `record of primitive` or `record of sequence`, or `record of record` etc.  Infact, this is why I wrote things from scratch instead of copying old tests for this parser because old tests are all "randomish".

And on the other hand this particular, this test is very specific - "an invalid case of a record having a value which is having single quotes",  such tests can be theoretically infinite. This opens door to having "random" tests to exist in golem-rib.